python/GafferUI/CompoundPlugValueWidget.py

```diff
@@ -199,20 +199,19 @@ def _childPlugs( self ) :
 	def __updateChildPlugUIs( self ) :
 
 		# ditch child uis we don't need any more
-		childNames = set( self.getPlug().keys() )
-		for childName in self.__childPlugUIs.keys() :
-			if childName not in childNames :
-				del self.__childPlugUIs[childName]
+		for childPlug in self.__childPlugUIs.keys() :
+			if childPlug not in self._childPlugs() :
```

Comment:
Let's store the result of a single call to `_childPlugs()` in a variable rather than call it over and over here (we can also reuse the same value on line 208). The `_childPlugs()` call is typically linear in the number of plugs, as is the outer loop, so by repeating the call we have quadratic complexity, which might get ugly with large plug counts.